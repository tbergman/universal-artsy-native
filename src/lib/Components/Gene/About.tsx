import React from "react"
import { createFragmentContainer, graphql } from "react-relay"

import { StyleSheet, View, ViewStyle } from "react-native"

import Biography from "./Biography"

import RelatedArtists from "../RelatedArtists"
import Separator from "../Separator"

import { About_gene } from "__generated__/About_gene.graphql"

interface Props {
  gene: About_gene
}

class About extends React.Component<Props> {
  render() {
    return (
      <View>
        {this.biography()}
        {this.relatedArtists()}
      </View>
    )
  }

  biography() {
    return (
      <View>
        <Biography gene={this.props.gene as any} style={styles.sectionSeparator} />
        <Separator style={styles.sectionSeparator} />
      </View>
    )
  }

  relatedArtists() {
    return (this.props.gene.trending_artists || []).length ? (
      <RelatedArtists artists={this.props.gene.trending_artists as any} />
    ) : null
  }
}

interface Styles {
  sectionSeparator: ViewStyle
}

const styles = StyleSheet.create<Styles>({
  sectionSeparator: {
    marginBottom: 20,
  },
})

export default createFragmentContainer(
  About,
  graphql`
    fragment About_gene on Gene {
      ...Biography_gene
      trending_artists {
        ...RelatedArtists_artists
      }
    }
  `
)
