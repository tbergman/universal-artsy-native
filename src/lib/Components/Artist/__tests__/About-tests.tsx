import "react-native"

import React from "react"
import * as renderer from "react-test-renderer"

import About from "../About"

it("renders properly", () => {
  const artist = {
    id: "banksy",
    has_metadata: true,
    is_display_auction_link: true,
    articles: [],
    related_artists: [],
  }
  const about = renderer.create(<About artist={artist} />).toJSON()
  expect(about).toMatchSnapshot()
})
