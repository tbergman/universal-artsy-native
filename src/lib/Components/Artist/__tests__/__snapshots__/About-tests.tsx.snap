// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders properly 1`] = `
<About
  artist={
    Object {
      "articles": Array [],
      "has_metadata": true,
      "id": "banksy",
      "is_display_auction_link": true,
      "related_artists": Array [],
    }
  }
/>
`;
